app/models/validators/beyond_osra_establishment_validator.rb

```diff
@@ -0,0 +1,13 @@
+class BeyondOsraEstablishmentValidator < ActiveModel::EachValidator
+  OSRA_ESTABLISHMENT_DATE = Date.new(2012,04,01)
+  include Helpers
+
+  def validate_each(record, attribute, value)
+    if !valid_date? value
+      record.errors[attribute] << ('is not a valid date')
```

Comment:
It can be reduced to
```ruby
if valid_date?(value) && (value < OSRA_ESTABLISHMENT_DATE
  record.errors[attribute] << ("is not valid (cannot be earlier than the OSRA establishment date of #{OSRA_ESTABLISHMENT_DATE})")
end
```